### DISCUSSION TITLE
WebUI 2.4.0

### DISCUSSION THREAD

**Discussion by hassandraga** [webui-dev/webui#163] [General]
@webui-dev/webui-maintainers-team.

Thank you guys for the remarkable contributions. This open source project is becoming better and better after every Git commit. I was busy with other things for a while, but now I will focus on WebUI, with your help we can release a new stable working version, 2.4.0, very soon.

I already read all the issues and discussions. I will push a new commit soon to combine all the new suggestions, while keeping WebUI simple, easy to build from source, bugs free as possible, and stay compatible with all the wrappers.

Thank you.

**Comment by ttytm** [comment]
When things are ready, should we release a `2.4.0` release candidate version `2.4.0-rc`?
Not marking it as latest, but as a pre-release version in github releases. 

We could already use it to prepare the wrappers on it and iron out stuff if something comes up.

Then, a few days later the `2.4.0` latest.

**Reply by hassandraga** [reply]
Sure. We can build a pre-release, and then when all wrappers work fine with it with documentation, then we build a final release version.

**Comment by hassandraga** [comment]
What still need to be finished to build the pre-release?

**Reply by ttytm** [reply]
Pre release is probably fine before having online documentation since the header file is fairly self documenting. 

But I would give it another week to finish things off and give wrappers some more days to catch up. Then a pre-release can be a good step to get more users start using the new version 🙂👍.

**Reply by hassandraga** [reply]
Sounds good

**Reply by ttytm** [reply]
One thing that fits before the pre-release:

I would update and make use of the release downloader in the go repo in the next days. It should replace the static lib files with downloads of webuis github releases, add debug and add better architecture support. 

You could then double check to make sure things are working. As a final step, I would then rebase the go repo to remove the static files from the history.

**Reply by hassandraga** [reply]
Sounds good. Thank you.

**Comment by hassandraga** [comment]
Is the main `C` lib ready to be released?

@webui-dev/webui-maintainers-team

**Reply by fibodevy** [reply]
I think it is, on Windows with Pascal wrapper it works fine

**Reply by hassandraga** [reply]
Unicode done.
Let's do more tests with wrappers. If all wrappers work fine then will release it.

**Comment by hassandraga** [comment]
- [x] Unicode
- [x] TLS
- [x] Multi-packet

**Reply by hassandraga** [reply]
Please, let's do some more tests to release it.

**Reply by fibodevy** [reply]
Im not C master, is there prebuilt WebUI TLS version? Or some instruction on how to build it myself? On Windows. Various errors shows up depending on what path to OpenSSL headers I provide. Eg there is include `configuration.h` and I have `configuration.h.in`, if I use other OpenSSL headers then there is error like "websocket should be included before windows".

**Reply by hassandraga** [reply]
I just added a section in the readme for the TLS build:
https://github.com/webui-dev/webui#build

**Reply by fibodevy** [reply]
@hassandraga `pascal-webui` has no tags and I cant add them, cant see the Gear icon/Settings tab, which I can see and edit for most other repos in this org, is this a matter of permission settings? If so, could you allow me to edit the pascal-webui repo's `About` to add some tags? So people could find this repo more easily on github.

**Reply by hassandraga** [reply]
@fibodevy Sure, I will check.

**Reply by hassandraga** [reply]
I just updated all repos settings for the whole team.

**Reply by fibodevy** [reply]
Thanks, already added some tags 🙂

**Comment by ttytm** [comment]
I had a free moment today and made an Odin binding I would add as another wrapper repo of the webui organisation if that's okay: https://github.com/webui-dev/odin-webui.

For now it supports the basic functionalities of the upcoming release, tested on mac and linux. It's set to private, as I didn't wanted to impose it as another wrapper. If it should be made public, the organisation owner would have to set it to public.

**Reply by hassandraga** [reply]
I like it. Thank you for creating this new wrapper 👍
Let's test it privately before we release WebUI to be sure the Odin wrapper can be used with the release.

**Reply by ttytm** [reply]
I had some trouble making it running with odin on windows. I'll give it another shot later this day 👍

**Comment by hassandraga** [comment]
I'm going to release v2.4.0 today. I will just finalize some texts and docs... then release it.

**Reply by hassandraga** [reply]
@webui-dev/webui-maintainers-team

**Reply by hassandraga** [reply]
@ttytm I don't know why the [release](https://github.com/webui-dev/webui/releases) does not include CI Linux pre-build. Maybe I missed something... Please feel free to delete/re-build/update the tags and settings to generate the full release build by CI.

**Reply by ttytm** [reply]
Added the linux archives that the workflow produced manually to the release.

Submitted a PR that should have them uploaded directly the next time a tag is created :+1:

**Reply by ttytm** [reply]
For the shield in the readme I guess we can use now the release version, then it won't swap back to `nightly` with the next change to the codebase.

[tag]: https://img.shields.io/github/v/tag/webui-dev/webui?style=for-the-badge&logo=webtrees&logoColor=C0CAF5&labelColor=414868&color=7664C6

[![][tag]](https://github.com/webui-dev/webui/releases/latest)

[release]: https://img.shields.io/github/v/release/webui-dev/webui?style=for-the-badge&logo=webtrees&logoColor=C0CAF5&labelColor=414868&color=7664C6
[![][release]](https://github.com/webui-dev/webui/releases/latest)

```diff
- https://img.shields.io/github/v/tag/webui-dev/webui?style=for-the-badge&logo=webtrees&logoColor=C0CAF5&labelColor=414868&color=7664C6
+ https://img.shields.io/github/v/release/webui-dev/webui?style=for-the-badge&logo=webtrees&logoColor=C0CAF5&labelColor=414868&color=7664C6
```

**Comment by ttytm** [comment]
Awesome progress with the new version. 

Congrats to the release @hassandraga! Thanks again for the work that went into this :100: :pray:

**Reply by hassandraga** [reply]
Thank to you @ttytm and to all contributors @webui-dev/webui-maintainers-team. Thank you for making this open-source project even better with each commit and issue resolved. I hope we can see decent apps using this latest release in the future.
